phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaExtractionProcessor.java

```diff
@@ -250,12 +247,31 @@ private void setHTableProperties(HTableDescriptor htd) {
         }
     }
 
-    private void setHColumnFamilyProperties(HColumnDescriptor columnDescriptor) {
-        Map<ImmutableBytesWritable, ImmutableBytesWritable> propsMap = columnDescriptor.getValues();
-        for (Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> entry : propsMap.entrySet()) {
+    private void setHColumnFamilyProperties(HColumnDescriptor[] columnDescriptors) {
+        Map<ImmutableBytesWritable, ImmutableBytesWritable> propsMap = columnDescriptors[0].getValues();
+        for(Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> entry : propsMap.entrySet()){
             ImmutableBytesWritable key = entry.getKey();
-            ImmutableBytesWritable value = entry.getValue();
-            definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
+            ImmutableBytesWritable defaultValue = entry.getValue();
+            Map<String, String> CFMap = new HashMap<String, String>();
+            Set<ImmutableBytesWritable> set = new HashSet<ImmutableBytesWritable>();
+            for(HColumnDescriptor columnDescriptor: columnDescriptors){
+                String columnFamilyName = Bytes.toString(columnDescriptor.getName());
+                ImmutableBytesWritable value = columnDescriptor.getValues().get(key);
+                // check if it is universal properties
+                if (SYNCED_DATA_TABLE_AND_INDEX_COL_FAM_PROPERTIES.contains(Bytes.toString(key.get()))){
+                    definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
+                    continue;
+                }
+                CFMap.put(columnFamilyName, Bytes.toString(value.get()));
+                set.add(value);
```

Comment:
I think we can maintain a map instead of [propertyValue and occurances] and execute line no. 273 only when exactly one entry from the map has occurrences equal to the no. of column families. 
Also, good to refactor this logic in another function.